### ISSUE TITLE
Web UI becomes unresponsive with large number of log entries

### ISSUE THREAD

**Issue by philwebb** [concourse/concourse#470]
When looking at build results that contain a large number of log lines the Web UI becomes unresponsive and often crashes the browser. I've tried both Firefox and Chrome.

For typical Java jobs it's quite common to have verbose build logging, and I'm not keen to reduce the amount of output since it's often beneficial when debugging a failed job.

It's quite hard to track down what might be causing the issue since the browser is totally unresponsive and the debugging tools don't work. However, looking at a shorter log I'm wondering if it's the sheer number of elements being created that's causing the problem. Each line results in a `div` and styled `span` element:

``` html
<div>
<span style="font-weight: normal; text-decoration: none; font-style: normal;" class="">7004cfc6e122: Pull complete</span>
</div>
```

For a large number lines this might be overwhelming the browser.

One possible solution might be to append lines into a single `<pre></pre>` element.

**Comment by philwebb** [comment]
[Here's som typical build output](https://build.spring.io/download/BOOT-PUB-JOB1/build_logs/BOOT-PUB-JOB1-4250.log) if it helps.

**Comment by philwebb** [comment]
One option that might help is to use something like the [Monaco Editor](https://github.com/Microsoft/monaco-editor) to show log output. It appears to handle large files quite well and could potentially defer color styling until the last minute and restrict it to just the visible part of the screen.

**Comment by jama-pivotal** [comment]
Sorry for the late response, but I think we could consider this closed with some of the work done in #1543, and with the follow up actions described there. Closing for now, but feel free to re-open if you feel i did this in error.
